osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -112,29 +113,57 @@ public WorkingBeatmap CreateNew(RulesetInfo ruleset, APIUser user)
         /// The new difficulty will be backed by a <see cref="BeatmapInfo"/> model
         /// and represented by the returned <see cref="WorkingBeatmap"/>.
         /// </summary>
-        public virtual WorkingBeatmap CreateNewBlankDifficulty(BeatmapSetInfo beatmapSetInfo, RulesetInfo rulesetInfo)
+        public virtual WorkingBeatmap CreateNewDifficulty(NewDifficultyCreationParameters creationParameters)
         {
-            // fetch one of the existing difficulties to copy timing points and metadata from,
-            // so that the user doesn't have to fill all of that out again.
-            // this silently assumes that all difficulties have the same timing points and metadata,
-            // but cases where this isn't true seem rather rare / pathological.
-            var referenceBeatmap = GetWorkingBeatmap(beatmapSetInfo.Beatmaps.First());
+            var referenceBeatmap = creationParameters.ReferenceBeatmap;
+            var targetBeatmapSet = creationParameters.BeatmapSet;
 
-            var newBeatmapInfo = new BeatmapInfo(rulesetInfo, new BeatmapDifficulty(), referenceBeatmap.Metadata.DeepClone());
+            BeatmapInfo newBeatmapInfo;
+            IBeatmap newBeatmap;
+
+            if (creationParameters.CreateBlank)
+            {
+                newBeatmapInfo = new BeatmapInfo(creationParameters.Ruleset, new BeatmapDifficulty(), referenceBeatmap.Metadata.DeepClone());
+                newBeatmap = new Beatmap { BeatmapInfo = newBeatmapInfo };
+                foreach (var timingPoint in referenceBeatmap.ControlPointInfo.TimingPoints)
+                    newBeatmap.ControlPointInfo.Add(timingPoint.Time, timingPoint.DeepClone());
+            }
+            else
+            {
+                newBeatmap = referenceBeatmap.Clone();
+
+                var referenceBeatmapInfo = referenceBeatmap.BeatmapInfo;
+                newBeatmap.BeatmapInfo = newBeatmapInfo = new BeatmapInfo(referenceBeatmapInfo.Ruleset, referenceBeatmapInfo.Difficulty.Clone(), referenceBeatmapInfo.Metadata.DeepClone())
+                {
+                    // Only selected appropriate properties are copied over.
+                    // Things like database ID, online status/ID, MD5 hash, star rating, etc. are omitted
+                    // because they should not be copied over and/or they will be recomputed on save.
```

Comment:
I know that any readers will probably raise an eyebrow at what follows. I don't like this either, but I don't know what the solution is. The fact of the matter is that some of `BeatmapInfo` must be copied, and some of it must not. It's just not a straight clone - a straight clone that keeps IDs / MD5 hashes / online status leads to various degrees of looking wrong or even corruption.

Initially I tried doing this the other way around (i.e. resetting all properties that needed to be reset after a full clone) but I kept missing stuff (see several commits that expand test coverage because I missed something that caused corruption). So I switched to opt-in rather than opt-out in the end.

Also note that length and star rating are not recomputed on save yet, which will lead to beatmap wedge looking a bit wrong. But that goes for any edited beatmap currently and would/should be resolved by recomputing on save.

Reply:
I was also going to suggest the other way around. Not sure how we can safely do it this way and ensure that we don't miss copying some new properties added in the future (I guarantee this is going to happen).

Reply:
The devil's advocate argument is that with opt-out somebody can add a property that shouldn't copy across and forget to reset it. At that point I guess it's a matter of which one is more likely, and I suppose your worry is more relevant than mine in that respect.

I'll revert to opt-out.

Reply:
Has been changed back to opt-out.